modules/core/02-client/keeper/client.go

```diff
@@ -17,7 +17,7 @@ func (k Keeper) CreateClient(
 	ctx sdk.Context, clientState exported.ClientState, consensusState exported.ConsensusState,
 ) (string, error) {
 	params := k.GetParams(ctx)
-	if !params.IsAllowedClient(clientState.ClientType()) {
+	if !params.IsAllowedClient(clientState.ClientType()) || clientState.ClientType() == exported.Localhost {
 		return "", sdkerrors.Wrapf(
 			types.ErrInvalidClientType,
 			"client state type %s is not registered in the allowlist", clientState.ClientType(),
```

Comment:
Maybe a nit, but wouldn't this error be a bit misleading for localhost? I mean, the reader of the error would see that localhost is in the allow list...

Reply:
Yes, I suppose it would! 

We can either change the error msg or put the localhost check in its own conditional, what do you think?

Reply:
Updated!